src/pyhf/tensor/tensorflow_backend.py

```diff
@@ -195,26 +186,20 @@ def reshape(self, tensor, newshape):
         return tf.reshape(tensor, newshape)
 
     def divide(self, tensor_in_1, tensor_in_2):
-        tensor_in_1 = self.astensor(tensor_in_1)
-        tensor_in_2 = self.astensor(tensor_in_2)
         return tf.divide(tensor_in_1, tensor_in_2)
 
     def log(self, tensor_in):
-        tensor_in = self.astensor(tensor_in)
         return tf.math.log(tensor_in)
 
     def exp(self, tensor_in):
-        tensor_in = self.astensor(tensor_in)
         return tf.exp(tensor_in)
 
     def stack(self, sequence, axis=0):
         return tf.stack(sequence, axis=axis)
 
     def where(self, mask, tensor_in_1, tensor_in_2):
-        mask = self.astensor(mask)
-        tensor_in_1 = self.astensor(tensor_in_1)
-        tensor_in_2 = self.astensor(tensor_in_2)
-        return mask * tensor_in_1 + (1 - mask) * tensor_in_2
+        mask = tf.cast(mask, tf.bool)
+        return tf.compat.v2.where(mask, tensor_in_1, tensor_in_2)
```

Comment:
Can we always use latest, or does `tf` only version this method?

Reply:
> Can we always use latest, or does `tf` only version this method?

@kratsg Sorry, I'm not quite sure what you mean by "latest" here, but if I guess a bit the reason I'm using `tf.compat.v2.where` is because it is [the alias](https://www.tensorflow.org/api_docs/python/tf/where) that TensorFlow recommends when you try to use `tf.where` while using the TF v1.0 API.